### PR TITLE
Fix condition in ILM step that cannot be met

### PR DESCRIPTION
**Pull request by cbuescher**
ReplaceDataStreamBackingIndexStep#performAction seems to perform an equality
check on an original Index and the write indexes names, but because this
compares an Index instance to a String, the condition can never be met.

I added a test that triggers this code path but am not sure if this scenario can
really happen in practice or if the test is too constructed. Happy to change,
just wanted to raise this since I happend to run across the warning in the code.

### CODE REVIEW

**Review by dakrone — APPROVED** [review]
LGTM, thanks for finding this, I think the test looks reasonable to me

**Review by dakrone — APPROVED** [review]
LGTM, thanks for finding this, I think the test looks reasonable to me

**Comment by dakrone** [comment]
Huh I don't know why my review was left twice, but at least it worked?

**Comment by cbuescher** [comment]
Thanks for the review @dakrone, do you think this should also go to 7.10?

**Review by andreidan — APPROVED** [review]
LGTM, thanks for fixing this Christoph.

Yes, this should go into 7.10 too. I believe we can look at this as a bug fix and port it to 7.9.2 as well

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ReplaceDataStreamBackingIndexStep.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public ClusterState performAction(Index index, ClusterState clusterState) {
         }
 
         assert dataStream.getWriteIndex() != null : dataStream.getName() + " has no write index";
-        if (dataStream.getWriteIndex().getIndex().equals(originalIndex)) {
+        if (dataStream.getWriteIndex().getIndex().getName().equals(originalIndex)) {
```
Comment (andreidan) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ReplaceDataStreamBackingIndexStep.java:78]: on second thought, to be consistent, I believe we should check indices equality (not just the name)
```suggestion
        if (dataStream.getWriteIndex().getIndex().equals(index)) {
```
We already do this in [another step](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/CheckNotDataStreamWriteIndexStep.java#L64)

**Review by andreidan — CHANGES_REQUESTED** [review]
Just a small change request Christoph, thank you

**Comment by cbuescher** [comment]
@andreidan thanks, I pushed the change you suggested

**Review by andreidan — APPROVED** [review]
LGTM, thanks for fixing this

**Pull request merged**